src/java.net.http/share/classes/jdk/internal/net/http/RequestPublishers.java

```diff
@@ -418,6 +418,11 @@ private int read() {
                 nextBuffer.position(0);
                 return n;
             } catch (IOException ex) {
+                need2Read = false;
+                haveNext = false;
+                try {
+                    is.close();
+                } catch (IOException ex2) {}
                 return -1;
```

Comment:
```
                } catch (IOException ex2) {}
                return -1;
```

I wonder if the first exception `ex` should actually be rethrown here instead of returning `-1`. Have you tried to explore this possibility?

Reply:
`read()` is not have IOE as checked exception, and also currently IOE is ignored.
So I ignored IOE at `close()` in this PR to minimize side-effect.

Reply:
Right. But I am not sure that is the right thing to do. If InputStream::read throws then it's likely that the request body will be missing some bytes, so the request should probably be cancelled/aborted at this point - rather than having a truncated body sent to the server.